PyriteEditor/runtime/res/shaders/depthOnly.fx

```diff
@@ -9,22 +9,48 @@ struct VertexInput
     float2 uv : UV;
 };
 
+struct VertexOut
+{
+    float4 position : SV_Position;
+    float4 worldPos : Position;
+};
+
 //////////////////////////////////////////////////////////////////////////////////////////////////
 
-float4 TransformVS(VertexInput vsIn) : SV_Position
+VertexOut DepthVS(VertexInput vsIn) 
 {
     float4x4 MVP = mul(ViewProj, ModelMatrix);
-    return mul(MVP, vsIn.Pos);
+    
+    VertexOut vso;
+    vso.position = mul(MVP, vsIn.Pos);
+    vso.worldPos = mul(ModelMatrix, vsIn.Pos);
```

Comment:
Do `worldPos = M*in.pos` and then `position = VP*worldPos` to avoid the extra matrix multiplication